mitodl/bootcamp-ecommerce
# Add support for sticky notifications

#### What are the relevant tickets?
Fixes #824 

#### What's this PR do?
Adds sticky footer notifications for success and error notifications. Site-wide notifications will still show up at the top like they did before. (Site-wide notification code was moved around but this is just a refactor.)

I started working on auto-dismiss success notifications but it looked non-trivial to implement in this PR so I left it out.

#### How should this be manually tested?
To fake success and failure messages, add this to `ApplicationDashboardPage.componentDidMount`:

      async componentDidMount() {
        const { applications, addSuccessNotification, addErrorNotification } = this.props
        if (applications) {
          const orderId = this.getOrderIdFromCybersourceParams()
          if (orderId) {
            await this.handleCybersourcePageLoad(orderId)
          }
        }
    
        // TODO: make sure to remove this in the PR
        addSuccessNotification({
          props: {
            text: "success!"
          }
        })
        addErrorNotification({
          props: {
            text: "oh no!"
          }
        })
      }

Make sure the notifications stick to the bottom as you scroll up and down. Also, open the drawer and verify that the notification is visible with the drawer behind it.

For CMS notifications, add a new `SiteNotification` from the Django admin. Then go to the dashboard to view the notifications.

#### Screenshots
![Screenshot from 2020-09-01 17-21-01](https://user-images.githubusercontent.com/863262/91907300-96979000-ec77-11ea-838b-ac3896d3a0a8.png)
![Screenshot from 2020-09-01 17-28-21](https://user-images.githubusercontent.com/863262/91908603-d52e4a00-ec79-11ea-92ec-7d77d5269432.png)

Updated screenshot with newer font:
![Screenshot from 2020-09-02 14-43-11](https://user-images.githubusercontent.com/863262/92023366-ad4aef00-ed2a-11ea-9d13-4f5b356d90b3.png)
